libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/NavigationViewRouteEngine.java

```diff
@@ -0,0 +1,203 @@
+package com.mapbox.services.android.navigation.ui.v5.route;
+
+import android.content.Context;
+import android.location.Location;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+
+import com.mapbox.api.directions.v5.models.DirectionsResponse;
+import com.mapbox.api.directions.v5.models.DirectionsRoute;
+import com.mapbox.api.directions.v5.models.RouteLeg;
+import com.mapbox.api.directions.v5.models.RouteOptions;
+import com.mapbox.geojson.Point;
+import com.mapbox.services.android.navigation.ui.v5.NavigationViewOptions;
+import com.mapbox.services.android.navigation.v5.navigation.MapboxNavigationOptions;
+import com.mapbox.services.android.navigation.v5.route.NavigationRouteEngine;
+import com.mapbox.services.android.navigation.v5.route.RouteEngineListener;
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;
+import com.mapbox.services.android.navigation.v5.utils.LocaleUtils;
+
+import java.util.List;
+import java.util.Locale;
+
+public class NavigationViewRouteEngine extends NavigationRouteEngine implements RouteEngineListener {
```

Comment:
Again is `Engine` the right word here? Maybe `Manager`? And I'm not sure I understand this whole hierarchy (`NavigationViewRouteEngine` -> `NavigationRouteEngine` -> `RouteEngine`)